src/js_realm.hpp

```diff
@@ -1607,6 +1607,12 @@ realm::Realm::Config RealmClass<T>::write_copy_to_helper(ContextType ctx, Object
     ObjectDefaultsMap defaults;
     ConstructorMap constructors;
     bool schema_updated = get_realm_config(ctx, args.count, args.value, config, defaults, constructors);
+
+    // TODO: remove when https://github.com/realm/realm-core/issues/5798 is fixed
+    if (config.sync_config && config.sync_config->flx_sync_requested) {
+        throw std::runtime_error("`writeCopyTo` does not support flexible sync");
```

Comment:
We could maybe word this a little more positively as I suspect this is quite frustrating to hit as a user... not sure if it's needed though
```suggestion
        throw std::runtime_error("`writeCopyTo` does not currently support flexible sync");
```

Reply:
I have updated the message - also in the test.